scripts/install.sh

```diff
@@ -0,0 +1,331 @@
+  {
+    command git clone --progress --branch "$ZINIT_BRANCH" \
+      "https://github.com/${ZINIT_REPO}" \
+      "${ZINIT_REPO_DIR_NAME}" 2>&1 | {
+      "${ZINIT_TMPDIR}/git-process-output.zsh" || cat;
+    }
+  } 2>/dev/null
+
+  zinit_checkout_ref
+
+  if [ -d "${ZINIT_REPO_DIR_NAME}" ]; then
+    echo_success "Zinit succesfully installed to " \
+                 "${COLOR_BOLD_GREEN}${ZINIT_INSTALL_DIR}"
+    echo_info "Zinit Version: ${COLOR_BOLD_GREEN}$(zinit_current_version)"
+  else
+    echo_error "Failed to install Zinit to ${COLOR_BOLD_YELLOW}${ZINIT_INSTALL_DIR}"
+  fi
+}
+
+# Modify .zshrc
+edit_zshrc() {
+  rc_update=1
+  if grep -E '(zinit|zplugin)\.zsh' "${ZSHRC}" >/dev/null 2>&1; then
+    echo_warn "${ZSHRC} already contains zinit commands. Not making any changes."
+    rc_update=0
+  fi
+
+  if [ $rc_update -eq 1 ]; then
+    echo_info "Updating ${ZSHRC} (10 lines of code, at the bottom)"
+    zinit_home_escaped="$(echo "${ZINIT_HOME}" | sed "s|$HOME|\$HOME|")"
+    command cat <<-EOF >> "$ZSHRC"
+
+### Added by Zinit's installer
+if [[ ! -f ${zinit_home_escaped}/${ZINIT_REPO_DIR_NAME}/zinit.zsh ]]; then
+    print -P "%F{33} %F{220}Installing %F{33}ZDHARMA-CONTINUUM%F{220} Initiative Plugin Manager (%F{33}${ZINIT_REPO}%F{220})…%f"
+    command mkdir -p "${zinit_home_escaped}" && command chmod g-rwX "${zinit_home_escaped}"
+    command git clone https://github.com/${ZINIT_REPO} "${zinit_home_escaped}/${ZINIT_REPO_DIR_NAME}" && \\
+        print -P "%F{33} %F{34}Installation successful.%f%b" || \\
+        print -P "%F{160} The clone has failed.%f%b"
+fi
+
+source "${zinit_home_escaped}/${ZINIT_REPO_DIR_NAME}/zinit.zsh"
+autoload -Uz _zinit
+(( \${+_comps} )) && _comps[zinit]=_zinit
+EOF
+  fi
+
+  unset rc_update zinit_home_escaped
+}
+
+query_for_annexes() {
+  zshrc_annex_file="$(mktemp)"
+  command cat <<-EOF >>"$zshrc_annex_file"
+
+# Load a few important annexes, without Turbo
+# (this is currently required for annexes)
+zinit light-mode for \\
+    zdharma-continuum/zinit-annex-as-monitor \\
+    zdharma-continuum/zinit-annex-bin-gem-node \\
+    zdharma-continuum/zinit-annex-patch-dl \\
+    zdharma-continuum/zinit-annex-rust
+
+EOF
+  # Ask user if we should add the annexes to their zshrc
+  # If NO_INPUT is set, but NO_ANNEXES is the annexes bit gets appended to the
+  # config (ie. default to yes if NO_INPUT, unless NO_ANNEXES)
+  reply=n
+  if [ -n "$NO_INPUT" ]
+  then
+    [ -z "$NO_ANNEXES" ] && reply=y
+  else
+    echo "${COLOR_PALE_MAGENTA}${COLOR_RESET} Would you like to add 4 useful plugins" \
+         "- the most useful annexes (Zinit extensions that add new" \
+         "functions-features to the plugin manager) to the zshrc as well?" \
+         "It will be the following snippet:"
+    command cat "$zshrc_annex_file"
+    # shellcheck disable=2059
+    printf "${COLOR_PALE_MAGENTA}${COLOR_RESET} Enter y/n and press Return: "
+    read -r reply
+  fi
+
+  if [ "$reply" = y ] || [ "$reply" = Y ]; then
```

Comment:
```suggestion
  if [[ "$reply" =~ ^(Y|y)$ ]]; then
```

Reply:
https://stackoverflow.com/a/3232082

Reply:
That's 2 bash-isms

shellcheck will call you out on that:

- https://github.com/koalaman/shellcheck/wiki/SC3010
- https://github.com/koalaman/shellcheck/wiki/SC3015